### PR TITLE
Add ARM support

### PR DESCRIPTION
**Pull request by maru-sama**
Based on sacd-ripper/sacd-ripper#98 make it possible to compile it on a
Raspberry PI. For now I only did a quick compile and run test and the
speed seems ok given the CPU being used.

### CODE REVIEW

File: tools/sacd_extract/CMakeLists.txt [hunk 1]
```diff
@@ -63,13 +63,19 @@ include_directories("../../libs/libid3")
...
-  NOT (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG"))
-  add_definitions(
-      -pipe
-      -Wall -Wextra -Wcast-align -Wpointer-arith -O3
-      -Wno-unused-parameter -msse2)
+if(NOT (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG"))
+  # Extra flags for arm, currently only tested on RPI
+  if(${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "arm*")
+    add_definitions(
+        -O3 -mfpu=neon -mcpu=native -mtune=native -mfloat-abi=hard)
```
Comment (setmind) [inline tools/sacd_extract/CMakeLists.txt]: This should be
`-pipe -Wall -Wextra -Wcast-align -Wpointer-arith -O3 -Wno-unused-parameter`
In other words, we just need to remove -msse2 for ARM.  -mfpu -mcpu -mtune -mfloat-abi were found to be a problem in other arm devices, and RPi is fine without these parameters.

File: libs/libdstdec/dst_init.c [hunk 2]
```diff
@@ -80,17 +80,27 @@ RT:  Richard Theelen, PDSL-labs Eindhoven <r.h.m.theelen@philips.com>
...
   {
     fprintf(stderr,"ERROR: not enough memory available!\n\n");
   }
+#endif
   return Array;
 }
 
 static void MemoryFree(void *Array) 
 {
+#ifdef __arm__
```
Comment (setmind) [inline libs/libdstdec/dst_init.c]: This should be 
`#if defined(__arm__) || defined(__aarch64__)`
to cover 64-bit ARM devices.

File: libs/libdstdec/dst_init.c [hunk 3]
```diff
@@ -80,17 +80,27 @@ RT:  Richard Theelen, PDSL-labs Eindhoven <r.h.m.theelen@philips.com>
 static void *MemoryAllocate(int NrOfElements, int SizeOfElement) 
 {
   void *Array;
-
+#ifdef __arm__
```
Comment (setmind) [inline libs/libdstdec/dst_init.c]: This should be
`#if defined(__arm__) || defined(__aarch64__)`
to cover 64-bit ARM devices.

File: tools/sacd_extract/CMakeLists.txt [hunk 4]
```diff
@@ -63,13 +63,19 @@ include_directories("../../libs/libid3")
...
+  # Extra flags for GCC for at least linux and macosx
+  elseif (CMAKE_COMPILER_IS_GNUCC OR (CMAKE_C_COMPILER_ID MATCHES "Clang") AND
+    NOT (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG"))
```
[the remark this answers is not in the document]
Comment (maru-sama) [reply tools/sacd_extract/CMakeLists.txt]: Yes apparently this was fixed in the local branch but not pushed. I will redo the complete logic anyway.

**Comment by maru-sama** [comment]
Have a look at the new commit. I would have liked to make a conditional to check for intel CPUs. Apparently this is not that simple so I did it the other way around and checked for not ARM.

**Comment by ArminiusTux** [comment]
Thank you maru-sama, sacd_extract compiled nicely on a RPi 3 B+ with Raspbian.

**Pull request merged**